pandas/core/frame.py

```diff
@@ -4321,21 +4321,41 @@ def combiner(x, y, needs_i8_conversion=False):
     def update(self, other, join='left', overwrite=True, filter_func=None,
                raise_conflict=False):
         """
-        Modify DataFrame in place using non-NA values from passed
-        DataFrame. Aligns on indices
+        Modify in place using non-NA values from another DataFrame.
+
+        Aligns on indices. There is no return value.
 
         Parameters
         ----------
         other : DataFrame, or object coercible into a DataFrame
+            Index should be similar to one of the columns in this one. If a
+            Series is passed, its name attribute must be set, and that will be
+            used as the column name in the resulting joined DataFrame.
         join : {'left'}, default 'left'
+            Only left join is implemented, keeping the index and columns of the
+            original object.
         overwrite : boolean, default True
-            If True then overwrite values for common keys in the calling frame
+            How to handle non-NA values for overlapping keys.
+
+            * True : overwrite values in `self` with values from `other`.
```

Comment:
I think we should try to avoid using 'self', as not necessarily every pandas user knows how classes are written. 

There is not always an ideal alternative, but I would use "original object" or "calling object" (or use DataFrame instead of object)